src/uct/api/uct.h

```diff
@@ -2065,33 +2066,84 @@ ucs_status_t uct_md_query(uct_md_h md, uct_md_attr_t *md_attr);
 
 /**
  * @ingroup UCT_MD
- * @brief Allocate memory for zero-copy sends and remote access.
+ * @brief UCT allocation parameters specification field mask
  *
- * Allocate memory on the memory domain. In order to use this function, MD
- * must support @ref UCT_MD_FLAG_ALLOC flag.
- *
- * @param [in]     md          Memory domain to allocate memory on.
- * @param [in,out] length_p    Points to the size of memory to allocate. Upon successful
- *                             return, filled with the actual size that was allocated,
- *                             which may be larger than the one requested. Must be >0.
- * @param [in,out] address_p   The address
- * @param [in]     flags       Memory allocation flags, see @ref uct_md_mem_flags.
- * @param [in]     name        Name of the allocated region, used to track memory
- *                             usage for debugging and profiling.
- * @param [out]    memh_p      Filled with handle for allocated region.
+ * The enumeration allows specifying which fields in @ref uct_mem_alloc_params_t
+ * are present.
  */
-ucs_status_t uct_md_mem_alloc(uct_md_h md, size_t *length_p, void **address_p,
-                              unsigned flags, const char *name, uct_mem_h *memh_p);
+typedef enum {
+    /** Enables @ref uct_mem_alloc_params_t::flags */
+    UCT_MEM_ALLOC_PARAM_FIELD_FLAGS          = UCS_BIT(0),
+
+    /** Enables @ref uct_mem_alloc_params_t::address */
+    UCT_MEM_ALLOC_PARAM_FIELD_ADDRESS        = UCS_BIT(1),
+
+    /** Enables @ref uct_mem_alloc_params_t::mem_type */
+    UCT_MEM_ALLOC_PARAM_FIELD_MEM_TYPE       = UCS_BIT(2),
+
+    /** Enables @ref uct_mem_alloc_params_t::mds */
+    UCT_MEM_ALLOC_PARAM_FIELD_MDS            = UCS_BIT(3),
+
+    /** Enables @ref uct_mem_alloc_params_t::name */
+    UCT_MEM_ALLOC_PARAM_FIELD_NAME           = UCS_BIT(4)
+} uct_mem_alloc_params_field_t;
 
 
 /**
  * @ingroup UCT_MD
- * @brief Release memory allocated by @ref uct_md_mem_alloc.
- *
- * @param [in]     md          Memory domain memory was allocated on.
- * @param [in]     memh        Memory handle, as returned from @ref uct_md_mem_alloc.
+ * @brief Parameters for allocating memory using @ref uct_mem_alloc
  */
-ucs_status_t uct_md_mem_free(uct_md_h md, uct_mem_h memh);
+typedef struct {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_mem_alloc_params_field_t. Fields not specified in this mask will
+     * be ignored
+     */
+    uint64_t                     field_mask;
+
+    /**
+     * Memory allocation flags, see @ref uct_md_mem_flags
```

Comment:
@yosefe we have to define default flags if user does not explicitly specify ? is it "0" or we want to go with read/write local access ?